frontend/src/metabase/common/hooks/use-entity-update/use-entity-update.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from "react";
+import type { Action } from "@reduxjs/toolkit";
+import { useStore } from "metabase/lib/redux";
+import { checkNotNull } from "metabase/core/utils/types";
+import { State } from "metabase-types/store";
+
+export interface EntityInfo<TId> {
+  id: TId;
+}
+
+export interface EntityQueryOptions<TId> {
+  entityId?: TId;
+}
+
+export interface UseEntityUpdateProps<
+  TId,
+  TEntity,
+  TEntityInfo extends EntityInfo<TId>,
+  TEntityData,
+> {
+  update: (entityInfo: TEntityInfo, updates: Partial<TEntityData>) => Action;
```

Comment:
Some update actions require more properties than only `id`, so it should be generic